src/image.tsx

```diff
@@ -178,55 +178,74 @@ export const ImageContent = React.forwardRef<HTMLImageElement, ImageProps>(
 
         const onLoadHandler = React.useCallback(() => {
             setIsError(false);
-            setIsLoading(false);
+            if (imageRef.current) {
+                imageRef.current.style.opacity = '1';
+            }
             setTimeout(() => {
                 setHideLoadingFallback(true);
             }, styles.FADE_IN_DURATION_MS);
 
             onLoad?.();
         }, [onLoad]);
 
-        const img = (
-            // https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/309
-            // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
-            <img
-                style={{
-                    ...(isLoading && withLoadingFallback ? {opacity: 0} : {opacity: 1}),
-                    boxSizing: 'border-box',
-                    border,
-                }}
-                ref={combineRefs(imageRef, ref)}
-                src={src}
-                className={classnames(
-                    styles.image,
-                    sprinkles({
-                        position: ratio !== 0 ? 'absolute' : 'static',
-                        borderRadius: props.circular
-                            ? '50%'
-                            : noBorderSetting
-                            ? undefined
-                            : vars.borderRadii.container,
-                    })
-                )}
-                alt={alt}
-                onError={() => {
-                    setIsError(true);
-                    setIsLoading(false);
-                    setHideLoadingFallback(true);
-                    onError?.();
-                }}
-                onLoad={onLoadHandler}
-            />
-        );
-
         React.useEffect(() => {
             // Needed because there is some race condition with SSR and onLoad events
+            // load event could be fired before the component is hydrated and mounted client side
             // https://github.com/facebook/react/issues/15446
             if (imageRef.current?.complete) {
                 onLoadHandler();
             }
         }, [onLoadHandler]);
 
+        const isLoading =
```

Comment:
Why isLoading should always be true in server side? If I understood correctly, that case is handled by the trick of listening to the load event in line 239.

Reply:
the initial state of the image is with the skeleton (loading), and the server should render that.
The purpose of the `<script>` trick is to transition from the "loading" (skeleton) state, to the "loaded" (image shown) state removing the opacity 0. But we can't do that until the image has finished loading.